exporter/exporterhelper/queued_retry.go

```diff
@@ -0,0 +1,196 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterhelper
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/jaegertracing/jaeger/pkg/queue"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+// QueuedSettings defines configuration for queueing batches before sending to the nextSender.
+type QueuedSettings struct {
+	// Disabled indicates whether to not enqueue batches before sending to the nextSender.
+	Disabled bool `mapstructure:"disabled"`
+	// NumWorkers is the number of consumers from the queue.
+	NumWorkers int `mapstructure:"num_workers"`
+	// QueueSize is the maximum number of batches allowed in queue at a given time.
+	QueueSize int `mapstructure:"queue_size"`
+}
+
+func CreateDefaultQueuedSettings() QueuedSettings {
+	return QueuedSettings{
+		Disabled:   false,
+		NumWorkers: 10,
+		QueueSize:  5000,
```

Comment:
Any guidance on why this is a good default? Is this suitable for production purposes, or for development purposes?

Reply:
This was copied from queuedretry processor. No idea why was decided on this number.

Reply:
Added explanation in code and readme.